src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -266,14 +274,22 @@ const AppFrame = ({ appUrl }: Props): ReactElement => {
   }
 
   useEffect(() => {
+    if (!appUrl) {
+      throw Error(URL_NOT_PROVIDED_ERROR)
+    }
+
     const loadApp = async () => {
       try {
         const app = await getAppInfoFromUrl(appUrl)
         setSafeApp(app)
       } catch (err) {
+        setAppLoadError(() => {
```

Comment:
`manifest.json` should be mandatory only for custom apps.

In general, we need the `manifest.json` file to get the app metadata (name, description, etc), BUT it's not required if the app is in the app list on our backend since we already have all the info there.

Reply:
So changed the approach guys and managing LegalDisclaimer in parent. Seems to work nicely. What do you think?